File: Vec.js
class Vec {
	constructor(x, y){
		this.x = x;
		this.y = y;
	}

	plus(vec){
		return new Vec(this.x + vec.x, this.y + vec.y)
	}

	minus(vec){
		return new Vec(this.x - vec.x, this.y - vec.y)
	}
  
	// see Pythagorean Theorem for distance formula,, from (0, 0)
	get length(){
		return Math.sqrt(this.x ** 2 + this.y ** 2)
	}

	static distance(coordsA, coordsB) {
		return Math.sqrt(
			(coordsB[0] - coordsA[0]) ** 2 +
			(coordsB[1] - coordsA[1]) ** 2
		)
	}
}

console.log(new Vec(1, 2).plus(new Vec(2, 3)));
console.log(new Vec(1, 2).minus(new Vec(2, 3)));
console.log(new Vec(3, 4).length);
console.log(Vec.distance([0, 0], [12, 5]));

File: Vec.py
import math
class Vec:
	def __init__(self, x, y):
		self.x = x
		self.y = y

	def __repr__(self):
		return f"Vec x: {self.x}, y: {self.y}"

	def plus(self, vec):
		return Vec(self.x + vec.x, self.y + vec.y)

	def minus(self, vec):
		return Vec(self.x - vec.x, self.y - vec.y)

	@property
	def length(self):
		return math.sqrt(self.x ** 2 + self.y ** 2)

	@staticmethod
	def distance(coordsA, coordsB):
		return math.sqrt(
			(coordsB[0] - coordsA[0]) ** 2 +
			(coordsB[1] - coordsA[1]) ** 2
		)

print(Vec(1, 2).plus(Vec(2, 3)))
print(Vec(1, 2).minus(Vec(2, 3)))
print(Vec(3, 4).length)
print(Vec.distance([0, 0], [12, 5]))
